### DISCUSSION TITLE
Novo no mundo Front-End

### DISCUSSION THREAD

**Discussion by rbeiro** [frontendbr/forum#1331] [Discussão]
Olá, bom, faz mais ou menos 2 meses que descobrir o HTML, CSS e o JavaScript, venho estudando por um curso on-line e simplesmente me apaixonei. Hoje estou com a dúvida mortal de saber os próximos passos para conseguir algum estágio ou algo do tipo. Conversei sobre com muitas pessoas ao meu redor porém a maioria é leiga sobre o assunto. Então, gostaria de saber se uma faculdade como Ciência da computação, (pública, e como moro em SP seria provavelmente USP, o que demanda um esforço) por exemplo, me ajudaria específicamente na área do Front-End, ou fazer um curso como o da Caelum, ou simplesmente estudar por conta própria, com livros, cursos on-line. Enfim, simplesmente não faço a menor ideia do que fazer e gostaria de uma opinião de alguém que entende minimamente do assunto.

Desde já, agradeço.

**Comment by Edvan27** [comment]
Se puder entrar numa faculdade de CC e ainda sendo federal, entre. Se puder fazer curso online na Caelum ou Alura faça, ambos caminhos são bem indicados, mas é importante você acima de tudo exercitar o que está estudando, sabe? Seja algo pago ou não. Vai com calma e devagar, começa por html, pega o css e depois compreenda bem o JS e tudo que escrever que seja código, jogue no seu git, será importante para futuramente você perceber o quanto evoluiu.

**Comment by giovannism20** [comment]
Acho que se você quer focar em front-end, talvez agora no início não seja uma das suas prioridades focar em uma faculdade e sim em se capacitar. Estude melhor qual seria/será a sua grade ainda mais de ciência da computação e provavelmente vai ver que não será tão introduzido a essa área de web (pelo menos, eu não vejo). Lugares como a Alura é um lugar excelente para aprender, muitos projetos e costumam comentar sobre coisas atuais e se verificar os planos do site, verá que eles agora têm a formação com projeto final.

**Comment by cauequeiroz** [comment]
Opa @grsilva11, beleza cara?

> Hoje estou com a dúvida mortal de saber os próximos passos para conseguir algum estágio ou algo do tipo.

A minha recomendação é você procurar uma vaga para Desenvolvedor Front-end Junior. Você mora em SP, por aqui o que mais tem é vaga para Junior. Eu mesmo quando comecei minha jornada eu estava estudando não fazia nem dois meses e consegui em uma agência para trabalhar com html e css.

Quanto mais você souber e quanto mais você conseguir demonstrar que sabe, mais facil será de conseguir isso. Você tem algum portfolio já? Se não tiver é uma boa fazer, porque na entrevista além de falar que sabe x, y ou z (igual os outros 415 desenvolvedores que apareceram antes de você) você vai poder abrir alguns sites que você mesmo fez e mostrar para quem está te contratando.

Você não precisa ser um jedi ainda, mas mostrar que já consegue produzir algo e que, principalmente, sabe se virar e estudar por conta própria o que precisa é uma qualidade extremamente bem vista no mercado.

> Então, gostaria de saber se uma faculdade como Ciência da computação, (pública, e como moro em SP seria provavelmente USP, o que demanda um esforço) por exemplo, me ajudaria específicamente na área do Front-End, ou fazer um curso como o da Caelum, ou simplesmente estudar por conta própria, com livros, cursos on-line. 

Cara, de inicio uma faculdade de ciencia da computação será um esforço absurdo que, para o que você quer hoje, não fará muito sentido. Estude em casa mesmo, aprenda bem as tecnologias mais básicas (html, css, javascript e git) e isso já será mais do que suficiente para você entrar no mercado.

No futuro, caso você realmente queira construir uma carreira no desenvolvimento de software, o conhecimento de Ciencia da Computação vai ser fundamental e a forma mais simples de conseguir isso é fazendo uma faculdade. Mas se você for auto didata também é possivel aprender por conta própria esses conteúdos.

Mas deixa isso la para frente, foca em entrar no mercado e começar a ter experiência prática como desenvolvedor. Você terá tempo para faz o que quiser depois.

---

É isso, espero ter ajudado com a minha opinião sobre o assunto, qualquer coisa só chamar!

**Comment by simaodeveloper** [comment]
**Bóra aprender com calma**

Faço das palavras do @cauequeiroz minhas, não precisa se preocupar com faculdade agora, foque em aprender a base do front-end (HTML, CSS e JAVASCRIPT), te recomendo procurar fazer o [freecodecamp](https://www.freecodecamp.org/), caso o inglês ainda for um impecilho, te oriento a ver os cursoso do [Guanabara no youtube](https://www.youtube.com/user/cursosemvideo), lá você aprenderá a base e coisas a mais.

Mais pra frente te recomendo ir atrás de faculdade, eu mesmo não quero fazer, estou estudando pelos vídeos da Univesp no youtube:

- [Ciclo Básico de Engenharia](https://www.youtube.com/user/univesptv/playlists?view=50&sort=dd&shelf_id=15);
- [Curso de Engenharia de Computação](https://www.youtube.com/user/univesptv/playlists?view=50&sort=dd&shelf_id=18);

Outra coisa muito importante, não se sinta sozinho, **auto-didata não é solo-didata**, nos encha de perguntas, até as que você achar a mais idiota, na fase de aprendizado náo existe pergunta idiota, existe pergunta!

Se passar por algum período de **ansiedade** e achar que está andando em circulos, abra seu coração aqui também, vamos te ajudar, pois todos nós passamos por esse processo de descoberta e entendimento sobre como funciona a nossa área, vai com calma, **passo por passo**

@grsilva11 bons estudos, e qualquer coisa faz outro post e me marca!

---------------------

Dica: aprenda uma coisa de cada vez, fraguimente em pequenas tarefas, passo a passo (**baby-steps**)

**Comment by ninetails** [comment]
a coisa que mais falo pra quem começa, ainda mais em São Paulo: Meetups a gente tem de monte :D participe!

Das comunidades, aconselho o mesmo (só de vir aqui creio que já foi um passo grande, já que forum dentro de Github não vejo muito como acessível a quem está iniciando).

Inglês é essencial, pelo menos o básico pois mesmo a galera sendo boa em disseminar conteúdo por aqui, ainda sim os melhores artigos ainda são em inglês.

Dado os dois últimos, aqui já tiveram vários tópicos sobre cursos, livros etc, aproveite :D

Faculdade é uma boa, mas encare que pra Frontend será muito mais um diferencial que essencial. Nossa área (graças) ainda não pede graduação, mas pra quem tá começando é o que vai ditar se vc vai largar lá na frente em relação a outros candidatos. E uma graduação em uma ótima instituição, mais ainda. Mesmo assim a graduação não vai te ensinar coisa alguma sobre frontend.

Mas se vc já quer trabalhar, não aconselho USP logo de cara caso for pra área de Computação, os cursos (IME USP) são bem mais puxados para academia e não muito para mercado de trabalho. E a alternativa: FATEC-SP é boa mas é bem defasada na minha opinião. Diria que pra começar, no seu caso em específico, uma faculdade boa privada seria boa pois já teria um fácil acesso para conseguir bons estágios em lugares bons.

Também é possível tentar algo sem estar cursando faculdade, tentando Junior direto, caso opte por cursos (Caelum/Alura ou outros). Tem uns (bastante gente) que viajam ao botar vagas, botando anos de experiência e tal, mas uma boa parte tem ciência que um Junior tá pra aprender.

Pense em USP caso queira seguir carreira docente ou coisas mais de louco tipo Data Science, Matemática Aplicada Computacional, Engenharia... E tome o tempo pra estudar, não pra trabalhar.

**Comment by camillodev** [comment]
Tem um repositório com otimas dicas pra quem ta começando na carreira de Front end, da uma olhada nele: [Carreira Front End](https://github.com/woliveiras/front-end-career/blob/master/translations/pt-br/README.md)
Sobre fazer ou não faculdade da uma olhada nessa issue #1184 e nas issues que foram referenciadas nela.
Na minha opnião, como seu principal objetivo parece ser conseguir um emprego na area agora, não recomendo faculdade, será o caminho mais longo. 

Pra quem é totalmente leigo na area de tecnologia eu recomendo fortemente os cursos de [Front End da Udacity](https://www.udacity.com/course/front-end-web-developer-nanodegree--nd001).

Eu fiz os cursos e não me arrependo, a unica coisa que você precisa ficar atento sobre esses cursos é que o conteudo dele não fica disponivel eternamente pra vc como outros cursos online, fica 1 ano dps que vc se forma, mas vale a pena da uma olhada na atualização dessas regras. Além disso, os cursos dizem que são em português mas so tem legenda em portugues e alguns artigos somente em inglês, mas nada do que copiar e colar no google tradutor não de pra vc se virar. 
E eles davam desconto nos webinars que eles faziam. Procura a agenda e espera caso va comprar o curso. Acho que eles não tem mais suporte nacional, se vc manja de inglês vai fundo pq os reviews fazem mta diferença, agora se não manja vai perder as reuniões com especialistas que ajudam mto pra conseguir emprego e tirar duvidas.

**Comment by SrVieira** [comment]
Opa irmão, tudo beleza?

Bem vindo ao nosso mundo haha, como já dito pelos amigos acima é aconselhável iniciar no básico (HTML, CSS e JS), existem vários handbooks/repositórios para cada um destes assuntos, vou compartilhar aqui embaixo os que tenho seguido nessa curta caminhada na área de Front End.

[HTML Referece](https://htmlreference.io): É um ótimo guia para quem não possui muita familiaridade com a tecnologia, possui as referencias de cada tag e qual a sua real utilização.
[Front End Handbook 2018](https://github.com/FrontendMasters/front-end-handbook-2018/blob/master/learning/html-css.md): Aqui temos um Handbook que abrange as tecnologias de HTML/CSS e algumas metodologias relacionadas.
[Airbnb JS](https://github.com/airbnb/javascript): Um guia de estilo para utilizar em seus javascripts <3
[Idiomatic CSS](https://github.com/necolas/idiomatic-css): Princípios para a criação de folhas de estilo padronizadas, consistentes e de fácil entendimento. 
[33 JS Concepts](https://github.com/leonardomso/33-js-concepts): Que traz uma série de conceitos e suas aplicações em seus códigos JS.

Em relação a faculdade ou curso, você tem que fazer uma análise crítica, quer focar no mercado interno? externo? ambos? Caso for no mercado interno, muitas empresas possuem aquela idéia de que é necessário uma faculdade/curso superior, enquanto no mercado externo isso não tem muita importância.

Caso queira mesmo se aprofundar na carreira leia o artigo [Carreira Front End](https://github.com/woliveiras/front-end-career/blob/master/translations/pt-br/README.md) como já proposto pelo @camillodev 

E por fim deixo aqui uma dica pra você, tenha disciplina e foco, se não possuir estes dois requisitos principais você não terá um bom aprendizado seja em uma faculdade ou em um curso. E sempre com nós aqui no fórum

@edit 

Estava esquecendo de uma coisa bem importante, o inglês <3. Se tu tem vontade de trabalhar remoto para alguma empresa na gringa ou ir trabalhar fora é essencial. vlw

**Comment by Juuhz** [comment]
Jovem, 

Já passei por essas dúvidas e irei compartilhar um pouco da minha vivência ( Sou Dev Full Stack à 6 anos e atualmente trabalho como Front-End Engineer na Oi ).

~ Alerta textão uHUEHuEHEUhE ~

## Começo de tudo: Orkut! 
É meio incomum, mas sim, foi lá que eu fiz meus primeiros códigos. No Orkut você podia personalizar seu perfil com algumas tags básicas de HTML, como negrito, itálico, cor, etc... Gostava de fazer essa dinâmica e comecei a pesquisar sobre o assunto. 

Foi ai que eu descobri o amor da minha vida: HTML! 

Lembro que nos meus primeiros projetos eu me sentia um artista, via aquela tela branca se transformando em blocos que se viravam algo, e isso me enchia os olhos. 

Eu nem sonhava em trabalhar com isso, mas era algo que roubava meu tempo, porquê passava horas e horas estudando e fazendo mini-projetinhos. 

Então acho que você já começou certo, você começou APAIXONADO. 

## Estudos / Cursos / Faculdades
Antes de eu começar a trabalhar de fato com isso, eu fiz pelo GOVERNO um curso técnico voltado a desenvolvimento ( Técnico em Informática ), pelo Senac. Este curso abriu meus olhos e me deu as ferramentas necessárias para começar minha saga nessa área. 

Então se você tiver a oportunidade de fazer um curso presencial, opite por ele, porquê: 
- Você realmente vai estudar.
- Te dará vários insigths e segurança para começar na área. 
- Em no máximo 1 ano você já vai estar preparado para o mercado de trabalho. 
- Ao final, você terá nível Técnico no currículo. 

Há 6 anos atrás os cursos onlines não eram muito difundidos, mas hoje estão no auge tendo uma gama absurda de conteúdos, oque é bem interessante. Eu particularmente uso os cursos apenas como apoio, aprendo mais por conta própria me desafiando, por exemplo:

Vamos supor que eu quero aprender Python: Primeiro eu vejo algumas aulas, vejo como funciona a questão teórica e depois crio desafios da minha cabeça e começo a praticar, por exemplo:

1) Como fazer um fetch e ler uma api em Python. 

Quando tenho dificuldade em algo, pesquiso e leio sobre, e assim vou aprendendo, aos poucos. Depois de um tempo, estou craque. Esse é meu método, não gosto de ficar acompanhando cursos onlines e copiando e colando oque os professores fazem.  

Faculdade: Cara, na minha opinião, é somente para pegar um papel... você aprende bem superficialmente as coisas e você não sai preparado para o mercado de trabalho, diferente de um curso técnico. Eu comecei a minha quando eu já era Dev Jr em uma agência de publicidade. 

Sempre que os professores perguntam quem já trabalha na área, pouco mais de meia dúzia levantam a mão em uma turma de 25~40 pessoas.

Ou seja, só com o diploma do curso técnico, eu já consegui ganhar um salário de 3K sendo Jr em uma empresa. Fiz a faculdade pela necessidade de cargos mais altos. 

Então minha dica é: Se você quer se colocar no mercado de trabalho mais rápido, faça um curso técnico. 

## Tecnologias

Sem dúvidas eu te recomendaria se aprofundar em React pelos motivos:
- Tem uma grande adesão no mercado. 
- Tem uma ponte para o mercado de aplicativos ( React Native ), oque te da mais autonomias para vagas. 
- Você irá aprender um Javascript mais avançado. 
- Tem vários viés de tecnologias que andam em conjuntos e que o mercado pede, como: Testes unitários, metodologia ágil, um pouco de orientação a objetos, etc... 
- Vai te facilitar aprender Node.js ( bem requisitado no mercado ). 
- etc... 

Bom... É isto, espero ter ajudado :) 

Ah: Deixe seu Linkedin sempre atualizado. 

Abraços Dev.

**Comment by ViniciusGularte** [comment]
A dica do @Juuhz  sobre curso técnico é muito boa, realmente ele te abre a mente, e cursos de pouca duração são bons pra ver se você realmente quer aquilo( ja vi pessoal saindo do ensino médio direto pra alguma faculdade de engenharia e depois de alguns anos vê que não é aquilo que  queria).

Faculdade é bom pra ti pegar um estagio por ae, sem falar que se você quiser entrar num cargo de gerência no futuro em alguma empresa grande, uma hora ou outra vai precisar, ( embora  frontend você realmente não vai aprender muita coisa na faculdade não :grimacing: ).

Mas segue estudando, faz curso, projeto, faculdade, arruma um job, vai experimentando as coisas, uma hora você vai achar o que realmente você quer se dedicar.

Como você ta estudando Javascript, te recomendo esses dois livros que ando lendo, [Eloquent Javascript](https://github.com/braziljs/eloquente-javascriptl) e [You Dont Know Js](https://github.com/cezaraugusto/You-Dont-Know-JS), te dão muita boa base, também quando você estiver estudando provavelmente vai dar de cara com React, Vue, Angular, e o caramba; por enquanto não foca neles, pode até dar uma olhada, mas procura aprender a base antes, isso fez muita falta pra mim atrás.
